tests/test_timer.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2018-2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include <pthread.h>
+#include "test_define.h"
+#include "utils/timer.h"
+
+int finite_thread(void *args) { return *(int *)args; }
+
+int infinite_thread(void *args) {
+  while (1) {
+    // Cancellation point
+    pthread_testcancel();
+  }
+  return *(int *)args;
+}
+
+void test_timer_finite(void) {
+  const struct itimerspec timeout = {.it_interval = {.tv_sec = 0, .tv_nsec = 0},
+                                     .it_value = {.tv_sec = 1, .tv_nsec = 0}};
+  int args = 7;
+
+  ta_timer_t *timer_id = ta_timer_start(&timeout, finite_thread, &args);
+  TEST_ASSERT(timer_id != NULL);
+  int ret;
+  ta_timer_stop(timer_id, (void **)&ret);
```

Comment:
`&ret` is the address of `ret` with an integer type but ta_timer_stop requires a pointer to pointer(stores an address of pointer).

Reply:
I'll change the type to `void *` instead

Reply:
Maybe declare `int *ret;` and call `ta_timer_stop(timer_id (void **) &ret)`.